include/boost/multiprecision/gmp.hpp

```diff
@@ -3635,35 +3696,36 @@ namespace Eigen
          IsSigned = std::numeric_limits<self_type>::is_specialized ? std::numeric_limits<self_type>::is_signed : true,
          RequireInitialization = 1,
       };
-      static Real epsilon()
+      static constexpr Real epsilon() noexcept
```

Comment:
Unfortunately we can't make this change: for variable precision types epsilon may vary at runtime, therefore this may **not** be `constexpr`, nor may it call `numeric_limits` (which may not even have a specialization if the type has variable precision).  Apologies, there may be a lack of testing on my part here: these members are documented requirements for Eigen, but I couldn't figure out which Eigen functions were using them :(

I used boost::math::tools::digits as a convenient shortcut, it's going to be tricky to do this correctly without it, though not impossible.

The other functions here have similar issues I'm afraid :(